baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/CatchBlockLogExceptionTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.errorprone.CompilationTestHelper;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CatchBlockLogExceptionTest {
+
+    private static final String errorMsg = "BUG: Diagnostic contains: "
+            + "Catch block contains log statements but thrown exception is never logged";
+
+    private CompilationTestHelper compilationHelper;
+
+    @Before
+    public void before() {
+        compilationHelper = CompilationTestHelper.newInstance(CatchBlockLogException.class, getClass());
+    }
+
+    @Test
+    public void testLogException() {
+        test(IllegalArgumentException.class, "log.info(\"hello\", e);", Optional.empty());
+        test(RuntimeException.class, "log.info(\"hello\", e);", Optional.empty());
+        test(Exception.class, "log.info(\"hello\", e);", Optional.empty());
+        test(Throwable.class, "log.info(\"hello\", e);", Optional.empty());
+    }
+
+    @Test
+    public void testLogExceptionNotLastArg() {
+        test(RuntimeException.class, "log.info(\"hello\", e, \"world\");", Optional.of(errorMsg));
+    }
+
+    @Test
+    public void testNoLogException() {
+        test(RuntimeException.class, "log.info(\"hello\");", Optional.of(errorMsg));
+    }
+
+    @Test
+    public void testNoLogStatement() {
+        test(RuntimeException.class, "// Do nothing", Optional.empty());
+    }
+
```

Comment:
I think it'd be good to add a test case like this, mostly for documentation purposes:
```
try {
  foo();
} catch (IOException e) {
  throw new RuntimeException("Failed to foo()", e);
}
```

Reply:
I wonder if we should have a similar check to ensure you're passing the exception as a cause?

Reply:
We could at least have it for the builtin exception types that people might be tempted to reach to in ways like `new RuntimeException()`